### DISCUSSION TITLE
How to cleanup consolidation job (jobtype = O) or any jobs that dont contain data

### DISCUSSION THREAD

**Discussion by bruno-at-bareos** [bareos/bareos#1971] [Ideas]
With this conversation we want to elaborate the best options how to cleanup (delete) those types of jobs. Beside consolidation (O), we have also Restore jobs (R) and all admin jobs (D).

Usually you include those jobs in your administrative duties, that you run from time to time.

What I'm personally doing on my installation every month:
- retrieve a list jobids from jobtype O,R,D whatever the jobstatus which are older than 90 days, and delete them.
  ```select jobid,job,jobstatus from job where type in ('O','R','D') and realendtime <= now()-'90 days'::interval;```
- retrieve a list of jobids from jobtype B,A which have jobstatus different from T,W (E,A,f,c) and older than 90 days, and delete them.
  ```select jobid,job,jobstatus from job where type in ('A','B') and jobstatus not in ('T','W') and realendtime <= now()-'90 days'::interval;```
(last point should also include Migration/Copy jobs to cover this type of usage)

@bareos/maintainers 

> Maybe it makes sense to keep the information which job created another job (i.e. consolidation -> virtual full or copy -> copied job), then we could.
> 
> clean up orphaned jobs (i.e. those where all created jobs were already purged)
> list jobs that were created (i.e. what full/incs were produced by a specific consolidation job)

We invite you to share what you expected or how do you manage from those type of jobs.

**Comment by k-w-d-e** [comment]
Would be nice to see some kind of 'retention policy' / or 'max amount of' for such jobs.

Having a nice list of AI jobs and looking at older jobs just to see an unlimited amount of (consolidation (O)) Jobs is weird.

I could understand that like Restore (R) Jobs might be wanna keept for a longer time as some kind of history/documentation/log but for O jobs is see no sense in that (under the assumption that they are not required after the consolidation itself).

**Reply by bruno-at-bareos** [reply]
The main use I have of them, is to know what happen at consolidation time. Which jobs are considered, excluded etc.
I clean them after two months (my AI has rewrite everything after 6 weeks).

As always, and you certainly know, we will consider any PR that might propose a enhancement on this subject. ;-)

**Reply by k-w-d-e** [reply]
Makes sense for history/documentation/log purposes but after some time some kind of automatic cleanup probably would be appreciated.

Sadly i am no programmer nor have the time for getting into programming so all i can give at the moment is feedback on enhancements and such things. :D

**Comment by bruno-at-bareos** [comment]
Maintenance tasks have never been so easy as now with the future bareos webui ...
You can start playing with with Bareos 25.1.0 technical preview, packages bareos-webui-new & bareos-webui-proxy ...
